### PR TITLE
Issue #14971: Add input file in indentation check for pattern matchin…

### LINKED ISSUE #14971: Add Check Support for Java 21 Pattern Matching for Switch Syntax: Indentation
child of #14961 
I have read check documentation: https://checkstyle.org/checks/misc/indentation.html#Indentation
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
PS D:\CS\test> cat src/Test.java                                                 
record Point(int x, int y) { }
record square(Point upperLeft, Point lowerRight) { }

public class Test {
    public void test() {
        Point p = new Point(1, 2);
        switch (p) {
            case Point(int x, int y)
            when x > 0 && y > 0 -> System.out.println(x + y); // violation
            case Point(int x, int y) -> System.out.println("error");
        }
    }
}
PS D:\CS\test> cat config.xml                                                    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
 <module name="TreeWalker">
    <module name="Indentation">
    </module>
 </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.14.2-all.jar -c config.xml  src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:9:13: 'case' child has incorrect indentation level 12, expected level should be 16. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```
---

**Describe what you expect in detail.**

I consider this a false positive. I want no violation on `when`

---

### PR DESCRIPTION
**Pull request by mahfouz72**
closes #14971

### CODE REVIEW

**Review by nrmancuso — APPROVED** [review]
Looks good.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationPatternMatchingForSwitch.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                    Point(int x, int y) //indent:20 exp:20
+                            when x > 0 && y > 0 -> System.out.println(x + y); //indent:28 exp:28
+            case Point(int x, int y) -> System.out.println("error"); //indent:12 exp:12
+        } //indent:8 exp:8
+        switch (p) { //indent:8 exp:8
+            case    //indent:12 exp:12
+                    Point(int x, int y) //indent:20 exp:20
+            when x > 0 && y > 0 -> System.out.println(x + y); //indent:12 exp:16 warn
+            case Point(int x, int y) -> System.out.println("error"); //indent:12 exp:12
+        } //indent:8 exp:8
```
Comment (rnveach) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationPatternMatchingForSwitch.java:56]: Add an input like: (indentation is intentional)
````
        switch (p) { //indent:8 exp:8
            case    //indent:12 exp:12
Point(int x, int y)
when
x > 0 && y > 0
->
System.out.println(x + y);
            case Point(int x, int y) -> System.out.println("error"); //indent:12 exp:12
        } //indent:8 exp:8
````

**Review by rnveach — COMMENTED** [review]
Just to be clear, we are only validating `LITERAL_WHEN` and things around it, correct?

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationPatternMatchingForSwitch.java (hunk 1 above)
Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationPatternMatchingForSwitch.java:56]: > Just to be clear, we are only validating `LITERAL_WHEN` and things around it, correct?

Yes



> Add an input like: (indentation is intentional)

done.

> System.out.println(x + y);

this is weird indent:0 expected:0

Comment (rnveach) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationPatternMatchingForSwitch.java:56]: > this is weird indent:0 expected:0

Exactly. :)

This is a bug. Most likely this line is not being validated.

You will only see issues with this check if you do bad indentation. You other examples were "too good".

After talking with @nrmancuso , move this to a new issue. We need to continue opening other issues and will probably return to indentation later.

Comment (rnveach) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationPatternMatchingForSwitch.java:56]: @mahfouz72 Is this done?

Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationPatternMatchingForSwitch.java:56]: https://github.com/checkstyle/checkstyle/issues/15098

**Review by rnveach — APPROVED** [review]

**Pull request merged**